periphery/periphery_report.bzl

```diff
@@ -0,0 +1,115 @@
+            ctx.executable.periphery_tool,
+        ] + periphery_file_inputs.runfiles.files.to_list(),
+        arguments = [args],
+        outputs = [output_file],
+        command = "{executable} scan $@ > {output_path}".format(
+            executable = ctx.executable.periphery_tool.path,
+            output_path = output_file.path,
+        ),
+        mnemonic = "GeneratePeripheryReport",
+    )
+    return [
+        DefaultInfo(
+            files = depset([output_file]),
+            runfiles = periphery_file_inputs.runfiles,
+        ),
+        PeripheryReportInfo(
+            report = output_file,
+        ),
+    ]
+
+def _collect_file_inputs(ctx):
+    runfiles = ctx.runfiles(
+        files = [
+            ctx.executable.periphery_tool,
+        ],
+    )
+    periphery_file_target_mapping_files = []
+    periphery_indexstore_files = []
+    srcs_files = []
+    for dep in ctx.attr.deps:
+        dep_runfiles = []
+        periphery_file_target_mapping_runfiles = ctx.runfiles(transitive_files = dep[PeripheryInfo].periphery_file_target_mapping)
+        periphery_file_target_mapping_files.extend(dep[PeripheryInfo].periphery_file_target_mapping.to_list())
+        dep_runfiles.append(periphery_file_target_mapping_runfiles)
+        periphery_indexstore_paths_depset = ctx.runfiles(transitive_files = dep[PeripheryInfo].periphery_indexstore)
+        periphery_indexstore_files.extend(dep[PeripheryInfo].periphery_indexstore.to_list())
+        dep_runfiles.append(periphery_indexstore_paths_depset)
+        srcs_depset = ctx.runfiles(transitive_files = dep[PeripheryInfo].srcs)
+        srcs_files.extend(dep[PeripheryInfo].srcs.to_list())
+        dep_runfiles.append(srcs_depset)
+        for dep_runfile in dep_runfiles:
+            runfiles = runfiles.merge(dep_runfile)
+    return struct(
+        runfiles = runfiles,
+        periphery_file_target_mapping_files = periphery_file_target_mapping_files,
+        periphery_indexstore_files = periphery_indexstore_files,
+        srcs_files = srcs_files,
+    )
+
+periphery_report = rule(
+    implementation = _periphery_report_impl,
+    doc = "Creates a periphery report for the given targets.",
+    attrs = {
+        "deps": attr.label_list(
+            aspects = [collect_periphery_info_aspect],
+            doc = "The targets to generate a periphery report from.",
+        ),
+        "report_exclude_globs": attr.string_list(
+            default = [],
+            doc = "A list of file globs to exclude from the report.",
+        ),
+        "periphery_additonal_args": attr.string_list(
+            default = [
+                "--retain-objc-accessible",
+                "--retain-assign-only-properties",
+            ],
+            doc = "A list additional arguments to pass to the periperhy invocation.",
+        ),
+        "format": attr.string(
+            default = "xcode",
+            values = [
+                "xcode",
+                "json",
+            ],
```

Comment:
A few more to add: https://github.com/peripheryapp/periphery/blob/master/Sources/Shared/OutputFormat.swift#L4-L9. Alternatively, we could remove this attr in favor of specifying the format via `periphery_additonal_args`.

Reply:
So It think it could make sense to keep this since the format does inform the extension of the created file. Do you know what extension each of these formats should use?